x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/GeoIpUpgradeIT.java

```diff
@@ -26,8 +26,8 @@ public void testGeoIpDownloader() throws Exception {
             assertBusy(() -> {
                 Response response = client().performRequest(new Request("GET", "_ingest/geoip/stats"));
                 String tasks = EntityUtils.toString(response.getEntity(), StandardCharsets.UTF_8);
-                // The geoip downloader doesn't actually do anything since there are no geoip processors:
-                assertThat(tasks, Matchers.containsString("failed_downloads\":0"));
+                // The geoip downloader should be executed since a geoip processors is present in behavioral analytics default pipeline:
+                assertThat(tasks, Matchers.containsString("failed_downloads\":1"));
```

Comment:
failure is expected since the download endpoint is configured this way in the `build.gradle` file:

```
setting 'ingest.geoip.downloader.endpoint', 'http://invalid.endpoint'
```

Reply:
Good catch!